modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -429,11 +426,21 @@ public CompletableFuture<Ignite> start(@Language("HOCON") @Nullable String cfg)
                                 fut -> new ConfigurationCatchUpListener(cfgStorage, fut, LOG)
                         );
 
-                        return CompletableFuture.allOf(notifyConfigurationListeners(), recoveryFuture);
+                        return notifyConfigurationListeners()
+                                .thenCompose(t -> {
+                                    // Deploy all registered watches because all components are ready and have registered their listeners.
+                                    try {
+                                        metaStorageMgr.deployWatches();
```

Comment:
Can you please explain why you need to transfer the watch deployment here?

Reply:
AFAIK, `deployWatches` should be called only after initial configuration notification listeners are notified